fabric-data-attachment-api-v1/src/main/java/net/fabricmc/fabric/api/attachment/v1/AttachmentTarget.java

```diff
@@ -0,0 +1,220 @@
+		A attached = getAttached(type);
+
+		if (attached != null) {
+			return attached;
+		} else {
+			A initialized = Objects.requireNonNull(initializer.get(), "initializer result cannot be null");
+			setAttached(type, initialized);
+			return initialized;
+		}
+	}
+
+	/**
+	 * Specialization of {@link #getAttachedOrCreate(AttachmentType, Supplier)}, but <i>only for attachment types with
+	 * {@link AttachmentType#initializer() initializers}.</i> It will throw an exception if one is not present.
+	 *
+	 * @param type the attachment type
+	 * @param <A>  the type of the data
+	 * @return the attached data, initialized if originally absent
+	 */
+	default <A> A getAttachedOrCreate(AttachmentType<A> type) {
+		Supplier<A> init = type.initializer();
+
+		if (init == null) {
+			throw new IllegalArgumentException("Single-argument getAttachedOrCreate is reserved for attachment types with default initializers");
+		}
+
+		return getAttachedOrCreate(type, init);
+	}
+
+	/**
+	 * Gets the data associated with the given {@link AttachmentType}, or returns the provided default value if it doesn't exist.
+	 * Unlike {@link #getAttachedOrCreate(AttachmentType, Supplier)}, this doesn't initialize the attachment with the default value.
+	 *
+	 * @param type         the attachment type
+	 * @param defaultValue the default value to use as fallback
+	 * @param <A>          the type of the attached data
+	 * @return the attached data, or the default value
+	 */
+	default <A> A getAttachedOrElse(AttachmentType<A> type, A defaultValue) {
+		Objects.requireNonNull(defaultValue, "default value cannot be null");
+
+		A attached = getAttached(type);
+		return attached == null ? defaultValue : attached;
+	}
+
+	/**
+	 * Gets the data associated with the given {@link AttachmentType}, or gets the provided default value from the
+	 * argument if it doesn't exist. Unlike {@link #getAttachedOrCreate(AttachmentType, Supplier)}, this doesn't
+	 * initialize the attachment with the default value.
+	 *
+	 * @param type         the attachment type
+	 * @param defaultValue the default value supplier to use as fallback
+	 * @param <A>          the type of the attached data
+	 * @return the attached data, or the default value
+	 */
+	default <A> A getAttachedOrGet(AttachmentType<A> type, Supplier<A> defaultValue) {
+		Objects.requireNonNull(defaultValue, "default value cannot be null");
+
+		A attached = getAttached(type);
+		return attached == null ? defaultValue.get() : attached;
+	}
+
+	/**
+	 * Sets the data associated with the given {@link AttachmentType}. Passing {@code null} removes the data.
+	 *
+	 * @param type  the attachment type
+	 * @param value the new value
+	 * @param <A>   the type of the data
+	 * @return the previous data
+	 */
+	@Nullable
+	default <A> A setAttached(AttachmentType<A> type, @Nullable A value) {
+		throw new UnsupportedOperationException("Implemented via mixin");
+	}
+
+	/**
+	 * Tests whether the given {@link AttachmentType} has any associated data.
```

Comment:
Please document the behavior for defaulted ones. Do they return true at all times or not?